src/main/java/com/microsoft/sqlserver/jdbc/SQLServerDatabaseMetaData.java

```
@@ -2537,6 +2537,22 @@ public boolean locatorsUpdateCopy() throws SQLException {
         checkClosed();
         return true;
     }
+
+    /* -------------- MSSQL-JDBC Extension methods  ---------------- */
+
+    public int getDatabaseCompatibilityLevel() throws SQLException {
+        checkClosed();
+        String database = connection.getCatalog();
+        try (SQLServerResultSet rs = getResultSetFromInternalQueries(null,
+                "select name, compatibility_level from sys.databases where name = '" + database + "'")) {
+            if (!rs.next()) {
+                return 0;
+            }
+            return rs.getInt("compatibility_level");
+        } catch (SQLServerException e) {
```

remove and just let the error throw

---

Fixed